cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -58,7 +58,9 @@ def _to_matrix_gate(matrix: npt.ArrayLike) -> cirq.MatrixGate:
 
 
 def counts_to_results(
-    counter: Dict[str, int], circuit: cirq.AbstractCircuit, param_resolver: cirq.ParamResolver
+    counter: Union[Dict[str, int], Dict[str, float]],
+    circuit: cirq.AbstractCircuit,
+    param_resolver: cirq.ParamResolver,
```

Comment:
ints should count as a floats
```suggestion
    counter: Dict[str, float], circuit: cirq.AbstractCircuit, param_resolver: cirq.ParamResolver
```

Reply:
> ints should count as a floats

On second thought (GitHub checks) `mypy` complains?

Reply:
It looks like that's coming from the interaction of this code with some other code here: https://github.com/Infleqtion/client-superstaq/blob/458a26a8670b1f2571c04114d7f745c80e7c1c3e/cirq-superstaq/cirq_superstaq/job.py#L179

Although mypy appears to be able to figure out that int literals can be treated like floats, it doesn't extend that logic to types contained within composite types. So in other words, treating an `int` as a `float` works, but not treating a `Dict[str, int]` like a `Dict[str, float]`.

Theoretically, I suppose that the solution here would be to propagate out the change to any code that uses this code, but that could potentially cause more problems than it solves. I think that going with the union of the two dict types makes sense here.

Reply:
Interesting, so I just explored this even more and figured out that the suggestion mentioned in the warning -- just changing `Dict` to `Mapping` actually fixes this completely as well. So you could just use `Mapping[str, float]` and be fine here.

Reply:
Oh, neat! Wonder why `Mapping` is treated differently than `Dict` -- seems like it shouldn't depend on the enclosing object type.

Done.

Reply:
`Mapping` is covariant and dict is invariant in value type (for this reason we should in general prefer `Mapping` over `Dict` for input annotations, and `dict` for output annotations)